zap/src/main/java/org/zaproxy/zap/extension/script/ScriptNode.java

```diff
@@ -97,4 +98,17 @@ public ScriptType getType() {
     public boolean isTemplate() {
         return template;
     }
+
+    @Override
+    public ScriptWrapper getUserObject() {
+        return script;
+    }
+
+    @Override
+    public void setUserObject(Object userObject) {
+        if (!(userObject instanceof ScriptWrapper script)) {
+            throw new IllegalArgumentException("Parameter userObject must be a ScriptWrapper.");
+        }
+        this.script = script;
```

Comment:
If you're concerned about the naming and hiding warning you mentioned, you could use `scriptWrap` or something

Reply:
I've approved anyway,  I don't think it's worth blocking on